src/components/taskDetails/index.tsx

```diff
@@ -0,0 +1,228 @@
+import React, {
+  ChangeEvent,
+  FC,
+  useEffect,
+  useContext,
+  useReducer,
+  useRef,
+  useState,
+} from 'react';
+import useFetch from '@/hooks/useFetch';
+import { isUserAuthorizedContext } from '@/context/isUserAuthorized';
+import NavBar from '@/components/navBar/index';
+import { toast, ToastTypes } from '@/helperFunctions/toast';
+import updateTaskDetails from '@/helperFunctions/updateTaskDetails';
+import task from '@/interfaces/task.type';
+import classNames from './task-details.module.scss';
+import TaskContainer from './TaskContainer';
+import Details from './Details';
+type Props = {
+  url: string;
+  taskID: string;
+};
+
+const initialState = {
+  taskDetails: {} as task,
+  editedDetails: {} as task,
+};
+
+function reducer(state: any, action: any) {
```

Comment:
Don't use `any` as type here and move this to other file named something like `currentFileName.reducer.ts`

Also check the below snippet for reference:

```ts
interface State {
  taskDetails: Record<string, any>;
  editedDetails: Record<string, any>;
}

interface Action {
  type: string;
  payload: Record<string, any>;
}

```

Reply:
Will address in next PR 👍🏻

Reply:
Updated as per the comment 👍🏻